src/dynamic_honey_badger/votes.rs

```diff
@@ -0,0 +1,302 @@
+        Ok(pk_opt.map_or(false, |pk| pk.verify(&signed_vote.sig, ser_vote)))
+    }
+}
+
+/// A vote fore removing or adding a validator.
+#[derive(Eq, PartialEq, Debug, Serialize, Deserialize, Hash, Clone)]
+struct Vote<NodeUid> {
+    /// The change this vote is for.
+    change: Change<NodeUid>,
+    /// The epoch in which the current era began.
+    era: u64,
+    /// The vote number: VoteCounter can be changed by casting another vote with a higher number.
+    num: u64,
+}
+
+/// A signed vote for removing or adding a validator.
+#[derive(Eq, PartialEq, Debug, Serialize, Deserialize, Hash, Clone)]
+pub struct SignedVote<NodeUid> {
+    vote: Vote<NodeUid>,
+    voter: NodeUid,
+    sig: Signature,
+}
+
+impl<NodeUid> SignedVote<NodeUid> {
+    pub fn era(&self) -> u64 {
+        self.vote.era
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeSet;
+    use std::rc::Rc;
+
+    use rand;
+
+    use super::{Change, SignedVote, VoteCounter};
+    use crypto::SecretKeySet;
+    use fault_log::{Fault, FaultKind, FaultLog};
+    use messaging::NetworkInfo;
+
+    /// Returns a vector of `node_num` `VoteCounter`s, and some signed example votes.
+    ///
+    /// If `signed_votes` is the second entry of the return value, then `signed_votes[i][j]` is the
+    /// the vote for `Remove(j)` by node `i`. Each node signed `Remove(0)`, `Remove(1)`, ... in
+    /// order.
+    fn setup(node_num: usize, era: u64) -> (Vec<VoteCounter<usize>>, Vec<Vec<SignedVote<usize>>>) {
+        let mut rng = rand::thread_rng();
+        let sk_set = SecretKeySet::random(3, &mut rng);
+        let ids: BTreeSet<usize> = (0..node_num).collect();
+        let pk_set = sk_set.public_keys();
+        let create_counter = |id: usize| {
+            let sk = sk_set.secret_key_share(id as u64);
+            let netinfo = NetworkInfo::new(id, ids.clone(), sk, pk_set.clone());
+            VoteCounter::new(Rc::new(netinfo), era)
+        };
+        let mut counters: Vec<_> = (0..node_num).map(create_counter).collect();
+        let sign_votes = |counter: &mut VoteCounter<usize>| {
+            (0..node_num)
+                .map(Change::Remove)
+                .map(|change| counter.sign_vote_for(change).expect("sign vote").clone())
+                .collect::<Vec<_>>()
+        };
+        let signed_votes: Vec<_> = counters.iter_mut().map(sign_votes).collect();
+        (counters, signed_votes)
+    }
+
+    #[test]
+    fn test_pending_votes() {
+        let node_num = 4;
+        let era = 5;
+        // Create the counter instances and the matrix of signed votes.
+        let (mut counters, sv) = setup(node_num, era);
+        // We will only use counter number 0.
+        let ct = &mut counters[0];
+
+        // Node 0 already contains its own vote for `Remove(3)`. Add two more.
+        let faults = ct
+            .add_pending_vote(&1, sv[1][2].clone())
+            .expect("add pending");
+        assert_eq!(faults, FaultLog::new());
```

Comment:
Maybe we should add an `is_empty()` method onto `FaultLog` so you can do `assert!(fault_log.is_empty());`?
```rust
impl<NodeUid: Clone> FaultLog<NodeUid> {
    pub fn is_empty(&self) -> bool {
        self.0.is_empty()
    }
}
```

Reply:
Same goes for all other `assert!(..., FaultLog::new());`.